Alderis/UIColor+HBAdditions.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  UIColor+HBAdditions.swift
+//  Alderis
+//
+//  Created by Ryan Nair on 10/5/20.
+//  Copyright © 2020 HASHBANG Productions. All rights reserved.
+//
+
+import UIKit
+
+extension UIColor {
+	
+	/// Initializes and returns a color object using data from the specified object.
+	///
+	/// The value is expected to be one of:
+	///
+	/// * An array of 3 or 4 integer RGB or RGBA color components, with values between 0 and 255 (e.g.
+	///   `@[ 218, 192, 222 ]`)
+	/// * A CSS-style hex string, with an optional alpha component (e.g. `#DAC0DE` or `#DACODE55`)
+	/// * A short CSS-style hex string, with an optional alpha component (e.g. `#DC0` or `#DC05`)
+	///
+	/// @param value The object to retrieve data from. See the discussion for the supported object
+	/// types.
+	/// @returns An initialized color object. The color information represented by this object is in the
+	/// device RGB colorspace.
+	
+	@objc convenience init?(hbcp_propertyListValue value: Any?) {
```

Comment:
Since Swift lets us get away with not having to namespace extensions, let’s make this:

```swift
@objc(initWithHbcp_propertyListValue:)
convenience init?(propertyListValue: Any?)
```

(Unfortunately the ugly objc method name has to match what we had before)

Additionally, we can be more explicit about supported types using a protocol:

```swift
protocol ColorValue: AnyObject {}

extension String: ColorValue {}
extension Array: ColorValue where Element: FixedWidthInteger {}

// and then…

@objc(initWithHbcp_propertyListValue:)
convenience init?(propertyListValue: ColorValue?)
```

`FixedWidthInteger` is the protocol `Int`, `UInt`, and a few others [conform to](https://developer.apple.com/documentation/swift/fixedwidthinteger#relationships). However I don’t know how that will fare with ObjC bridging. If you have trouble here, just leave it as `Any?` and I’ll see if I can take a stab at it sometime.